wagtail/api/v2/endpoints.py

```diff
@@ -88,88 +93,198 @@ def handle_exception(self, exc):
+        if fields_config and fields_config[0][0] == '*':
+            fields = set(all_fields)
+            fields_config = fields_config[1:]
+        elif fields_config and fields_config[0][0] == '_':
+            fields = set()
+            fields_config = fields_config[1:]
+
+        mentioned_fields = set()
+        sub_fields = {}
+
+        for field_name, negated, field_sub_fields in fields_config:
+            if negated:
+                try:
+                    fields.remove(field_name)
+                except KeyError:
+                    pass
             else:
-                fields = set(self.get_default_fields(model))
+                fields.add(field_name)
+                if field_sub_fields:
+                    sub_fields[field_name] = field_sub_fields
+
+            mentioned_fields.add(field_name)
+
+        unknown_fields = mentioned_fields - set(all_fields)
+
+        if unknown_fields:
+            raise BadRequestError("unknown fields: %s" % ', '.join(sorted(unknown_fields)))
+
+        # Build nested serialisers
+        child_serializer_classes = {}
+
+        for field_name in fields:
+            try:
+                django_field = model._meta.get_field(field_name)
+            except FieldDoesNotExist:
+                django_field = None
 
-            unknown_fields = fields - set(all_fields)
+            if django_field and django_field.is_relation:
+                child_sub_fields = sub_fields.get(field_name, [])
 
-            if unknown_fields:
-                raise BadRequestError("unknown fields: %s" % ', '.join(sorted(unknown_fields)))
+                # Inline (aka "child") models should display all fields by default
+                if isinstance(getattr(django_field, 'field', None), ParentalKey):
+                    if not child_sub_fields or child_sub_fields[0][0] not in ['*', '_']:
+                        child_sub_fields = list(child_sub_fields)
+                        child_sub_fields.insert(0, ('*', False, None))
 
-            # Reorder fields so it matches the order of all_fields
-            fields = [field for field in all_fields if field in fields]
+                # Get a serializer class for the related object
+                child_model = django_field.related_model
+                child_endpoint_class = router.get_model_endpoint(child_model)
+                child_endpoint_class = child_endpoint_class[1] if child_endpoint_class else BaseAPIEndpoint
+                child_serializer_classes[field_name] = child_endpoint_class._get_serializer_class(router, child_model, child_sub_fields, nested=True)
+
+            else:
+                if field_name in sub_fields:
+                    # Sub fields were given for a non-related field
+                    raise BadRequestError("'%s' does not support nested fields" % field_name)
+
+        # Reorder fields so it matches the order of all_fields
+        fields = [field for field in all_fields if field in fields]
```

Comment:
You can pick any beat on this [page](https://soundcloud.com/freehiphopbeatsforyou) and try to read this line. I bet you'll get a good rap song! :)